library-front/src/components/Header/Header.tsx

```diff
@@ -1,26 +1,50 @@
-import { FC } from 'react'
+import { FC, useEffect, useState } from 'react'
 import { Link } from 'react-router-dom'
 import { clearJwt, Jwt } from '../../helpers/jwt-helper'
+import BookSearch from '../BookSearch/BookSearch'
 import './Header.css'
 
 interface HeaderProps {
   jwt: Jwt | null
   setJwt: (jwt: Jwt | null) => void
+  setSearch: (search: string) => void
 }
 
-const Header: FC<HeaderProps> = ({ jwt, setJwt }) => {
+const Header: FC<HeaderProps> = ({ jwt, setJwt, setSearch }) => {
+  const [isHeaderVisible, setIsHeaderVisible] = useState(true)
+  const [prevScrollPos, setPrevScrollPos] = useState(0)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const currentScrollPos = window.pageYOffset
+
+      setIsHeaderVisible(prevScrollPos > currentScrollPos || currentScrollPos < 50)
```

Comment:
does this mean we have to scroll way back to the top to have visible header?

Reply:
No, the prevScrollPos > currentScrollPos means that when the user starts scrolling up, the header becomes visible